### ISSUE TITLE
bug: `sponsorblock` patch fails when certain patches are excluded

### ISSUE THREAD

**Issue by anil-al** [inotia00/ReVanced_Extended#1037]
### Type

Error while patching

### Tools used

rvx-builder

### Application

Youtube 18.24.37 in Android Termux

### Bug description

Sponsorblock patch gives the following error.

### Steps to reproduce

-

### Relevant log output

```shell
[builder] INFO: Reading dex files
[builder] INFO: Decoding AndroidManifest.xml only, because resources are not needed
[builder] INFO: Skipping bypass-ambient-mode-restrictions: Manually excluded
[builder] INFO: Skipping custom-branding-icon-mmt: Manually excluded
[builder] INFO: Skipping custom-branding-icon-revancify-blue: Manually excluded
[builder] INFO: Skipping custom-branding-icon-revancify-red: Manually excluded
[builder] INFO: Skipping custom-branding-youtube-name: Manually excluded
[builder] INFO: Skipping custom-package-name: Manually excluded
[builder] INFO: Skipping default-video-speed: Manually excluded
[builder] INFO: Skipping disable-haptic-feedback: Manually excluded
[builder] INFO: Skipping enable-tablet-mini-player: Manually excluded
[builder] INFO: Skipping enable-tablet-navigation-bar: Manually excluded INFO: Skipping enable-wide-search-bar: Manually excluded
[builder] INFO: Skipping force-hide-player-button-background: Manually excluded
[builder] INFO: Skipping force-vp9-codec: Manually excluded
[builder] INFO: Skipping hide-captions-button: Manually excluded
[builder] INFO: Skipping hide-category-bar: Manually excluded
[builder] INFO: Skipping hide-description-components: Manually excluded
[builder] INFO: Skipping hide-double-tap-overlay-filter: Manually excluded
[builder] INFO: Skipping hide-endscreen-cards: Manually excluded INFO: Skipping hide-endscreen-overlay: Manually excluded INFO: Skipping hide-filmstrip-overlay: Manually excluded INFO: Skipping hide-floating-microphone: Manually excluded INFO: Skipping hide-fullscreen-panels: Manually excluded INFO: Skipping hide-info-cards: Manually excluded
[builder] INFO: Skipping hide-live-chat-button: Manually excluded
[builder] INFO: Skipping hide-player-button-background: Manually excluded
[builder] INFO: Skipping hide-player-overlay-filter: Manually excluded
[builder] INFO: Skipping hide-previous-next-button: Manually excluded
[builder] INFO: Skipping hide-seekbar: Manually excluded
[builder] INFO: Skipping materialyou: Manually excluded
[builder] INFO: Skipping theme: Manually excluded
[builder] INFO: Skipping translations: Manually excluded
[builder] INFO: Merging /data/data/com.termux/files/home/rvx-builder/revanced/revanced-integrations-v0.111.2.apk
[builder] INFO: Deleting existing resource cache directory
[builder] INFO: Decoding resources
[builder] INFO: Executing patches
[builder] INFO: add-splash-animation succeeded
[builder] INFO: change-homepage succeeded
[builder] INFO: custom-double-tap-length succeeded
[builder] INFO: custom-seekbar-color succeeded
[builder] INFO: custom-video-speed succeeded
[builder] INFO: default-video-quality succeeded
[builder] INFO: disable-auto-captions succeeded
[builder] INFO: disable-hdr-video succeeded
[builder] INFO: disable-landscape-mode succeeded
[builder] INFO: disable-quic-protocol succeeded
[builder] INFO: disable-startup-shorts-player succeeded
[builder] INFO: enable-compact-controls-overlay succeeded
[builder] INFO: enable-debug-logging succeeded
[builder] INFO: enable-external-browser succeeded
[builder] INFO: enable-minimized-playback succeeded
[builder] INFO: enable-new-splash-animation succeeded
[builder] INFO: enable-new-thumbnail-preview succeeded
[builder] INFO: enable-old-quality-layout succeeded
[builder] INFO: enable-open-links-directly succeeded
[builder] INFO: enable-seekbar-tapping succeeded
[builder] INFO: enable-timestamps-speed succeeded
[builder] INFO: force-opus-codec succeeded
[builder] INFO: force-premium-heading succeeded
[builder] INFO: header-switch succeeded
[builder] INFO: hide-account-menu succeeded
[builder] INFO: hide-auto-player-popup-panels succeeded
[builder] INFO: hide-autoplay-button succeeded
[builder] INFO: hide-autoplay-preview succeeded
[builder] INFO: hide-button-container succeeded
[builder] INFO: hide-cast-button succeeded
[builder] INFO: hide-channel-avatar-section succeeded
[builder] INFO: hide-channel-watermark succeeded
[builder] INFO: hide-collapse-button succeeded
[builder] INFO: hide-comment-component succeeded
[builder] INFO: hide-crowdfunding-box succeeded
[builder] INFO: hide-email-address succeeded
[builder] INFO: hide-feed-flyout-panel succeeded
[builder] INFO: hide-general-ads succeeded
[builder] INFO: hide-layout-components succeeded
[builder] INFO: hide-load-more-button succeeded
[builder] INFO: hide-mix-playlists succeeded
[builder] INFO: hide-music-button succeeded
[builder] INFO: hide-navigation-buttons succeeded
[builder] INFO: hide-navigation-label succeeded
[builder] INFO: hide-pip-notification succeeded
[builder] INFO: hide-player-flyout-panel succeeded
[builder] INFO: hide-quick-actions succeeded
[builder] INFO: hide-seek-message succeeded
[builder] INFO: hide-shorts-component succeeded
[builder] INFO: hide-snack-bar succeeded
[builder] INFO: hide-speed-overlay succeeded
[builder] INFO: hide-suggested-actions succeeded
[builder] INFO: hide-suggestions-shelf succeeded
[builder] INFO: hide-time-stamp succeeded
[builder] INFO: hide-tooltip-content succeeded
[builder] INFO: hide-trending-searches succeeded
[builder] INFO: hide-video-ads succeeded
[builder] INFO: language-switch succeeded
[builder] INFO: layout-switch succeeded
[builder] INFO: microg-support succeeded
[builder] INFO: optimize-resource succeeded
[builder] INFO: overlay-buttons succeeded
[builder] INFO: return-youtube-dislike succeeded
[builder] INFO: settings succeeded
[builder] SEVERE: sponsorblock failed:
[builder] app.revanced.patcher.patch.PatchResultError: 'sponsorblock' depends on 'sponsorblock-bytecode-patch' but the following error was raised: java.lang.NullPointerException at app.revanced.patches.youtube.utils.sponsorblock.bytecode.patch.SponsorBlockBytecodePatch.execute(SponsorBlockBytecodePatch.kt:83) at app.revanced.patches.youtube.utils.sponsorblock.bytecode.patch.SponsorBlockBytecodePatch.execute(SponsorBlockBytecodePatch.kt:44) at app.revanced.patcher.Patcher.executePatches$executePatch(Patcher.kt:333) at app.revanced.patcher.Patcher.executePatches$executePatch(Patcher.kt:309) at app.revanced.patcher.Patcher.access$executePatches$executePatch(Patcher.kt:38) at app.revanced.patcher.Patcher$executePatches$1.invokeSuspend(Patcher.kt:358) at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33) at kotlin.sequences.SequenceBuilderIterator.hasNext(SequenceBuilder.kt:129) at app.revanced.utils.patcher.PatcherKt.applyPatchesVerbose(Patcher.kt:84) at app.revanced.cli.patcher.Patcher.start$revanced_cli(Patcher.kt:19) at app.revanced.cli.command.MainCommand.run(MainCommand.kt:172) at picocli.CommandLine.executeUserObject(CommandLine.java:2026) at picocli.CommandLine.access$1500(CommandLine.java:148) at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461) at picocli.CommandLine$RunLast.handle(CommandLine.java:2453) at picocli.CommandLine$RunLast.handle(CommandLine.java:2415) at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273) at picocli.CommandLine$RunLast.execute(CommandLine.java:2417) at picocli.CommandLine.execute(CommandLine.java:2170) at app.revanced.cli.main.MainKt.main(Main.kt:7) at app.revanced.patcher.Patcher.executePatches$executePatch(Patcher.kt:312) at app.revanced.patcher.Patcher.access$executePatches$executePatch(Patcher.kt:38) at app.revanced.patcher.Patcher$executePatches$1.invokeSuspend(Patcher.kt:358) at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33) at kotlin.sequences.SequenceBuilderIterator.hasNext(SequenceBuilder.kt:129) at app.revanced.utils.patcher.PatcherKt.applyPatchesVerbose(Patcher.kt:84) at app.revanced.cli.patcher.Patcher.start$revanced_cli(Patcher.kt:19) at app.revanced.cli.command.MainCommand.run(MainCommand.kt:172) at picocli.CommandLine.executeUserObject(CommandLine.java:2026) at picocli.CommandLine.access$1500(CommandLine.java:148) at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461) at picocli.CommandLine$RunLast.handle(CommandLine.java:2453) at picocli.CommandLine$RunLast.handle(CommandLine.java:2415) at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273) at picocli.CommandLine$RunLast.execute(CommandLine.java:2417) at picocli.CommandLine.execute(CommandLine.java:2170) at app.revanced.cli.main.MainKt.main(Main.kt:7)
[builder] INFO: spoof-app-version succeeded
[builder] INFO: swipe-controls succeeded
[builder] INFO: Compiling resources
[builder] WARNING: aapt2 W 07-05 18:44:13 17734 17734] Unknown chunk type '200'.
[builder] INFO: Writing modified dex files
[builder] INFO: Aligning com.google.android.youtube.apk to revanced_aligned.apk
[builder] INFO: Signing revanced_aligned.apk to revanced_signed.apk
[builder] INFO: Found existing keystore: revanced.keystore
[builder] INFO: Copying revanced_signed.apk to revanced.apk
[builder] INFO: Finished
[builder] Copied files over to /storage/emulated/0/! Please install ReVanced, its located in /storage/emulated/0/ReVanced-YouTube-cli_v2.21.6-patches_v2.181.1.apk and if you are building YT/YTM ReVanced without root, also install /storage/emulated/0/microg.apk.
```

### Device Environment

Android 13, Samsung Tab S8+

### Acknowledgements

- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.
- [X] I have written the title and contents in English.
- [X] This issue does not reproduce on unpatched YouTube or YT Music.

**Comment by sdsucks** [comment]
sponsoblock patch fails also on windows.

Youtube v18.24.37

**Comment by inotia00** [comment]
fixed in [revanced-patches-v2.181.4](https://github.com/inotia00/revanced-patches/releases/tag/v2.181.4)
